### ISSUE TITLE
Identifiyng the publications in citations co-couples

### ISSUE THREAD

**Issue by franastorga** [massimoaria/bibliometrix#443]
I am interested in identifyng with wich references one particular author its coupled. For example, I am interested in the author Juan Carver (not real), and I would like to know all the other references that are cited in the papers in which that author its cited. I used the co-citation tool trying to export the results, thinking that then I could filter only those couples with Juan Carver; however, I was not able to extract paper-by-paper, and export all the references that are co-citing Juan Carver. I could use a list of references that are cited within papers that cite Juan Carver or, alternatively, a list od all the publications that cite Juan Carver detailing all the references included by publication. 
Is this possible?

Similalry, I would like to identifty the publications in which two terms are coupled. For example, if there is a frequent mention of the words TREE and FLOWER, I would like to obtain a list of all the publications in which those terms are coupled (i.e., papers that mention those terms). Is this possible?
Thanks!!

**Comment by massimoaria** [comment]
Yes, both are possible today, and neither needs the network functions. You were reaching for the
co-citation network, but the network is the *aggregated* result -- what you want is the object it is
aggregated from, and that object is exported.

`cocMatrix()` returns a document-by-item incidence matrix: one row per document of your collection,
one column per item of the chosen field. Both of your questions are a subsetting of that matrix.

## Publications where two terms are coupled

```r
library(bibliometrix)

WK <- cocMatrix(M, Field = "ID", sep = ";")   # "DE" for author keywords
both <- rownames(WK)[WK[, "TREE"] > 0 & WK[, "FLOWER"] > 0]
both
```

`both` is the list of documents mentioning both terms. Note that the column names are upper case and
exactly as they appear in the field, so `colnames(WK)` or `grep("TREE", colnames(WK), value = TRUE)`
is the way to find the right spelling first.

## References co-cited with a given author

Same matrix, built on the cited references:

```r
WCR <- cocMatrix(M, Field = "CR", sep = ";")

# 1. locate the cited work(s) of the author, by surname
cols <- grep("CARVER J", colnames(WCR), value = TRUE)

# 2. the documents of your collection that cite them
citing <- rownames(WCR)[rowSums(WCR[, cols, drop = FALSE]) > 0]

# 3. everything those documents cite, ranked -- the works co-cited with the author
co <- colSums(WCR[citing, , drop = FALSE])
co <- sort(co[co > 0 & !(names(co) %in% cols)], decreasing = TRUE)
head(data.frame(reference = names(co), n = as.integer(co)), 20)
```

That gives you both of the alternatives you described: `citing` is the list of publications citing
the author, and `co` is the list of references co-cited with them, with the number of documents each
was co-cited in. For the references of one specific citing paper:

```r
colnames(WCR)[WCR["SMITH J, 2020, J SOMETHING", ] > 0]
```

Run against the package's own sample collection, step 1 matches one cited work, step 2 finds the 5
documents citing it, and step 3 returns the 153 references co-cited with it, ranked.

## One caveat

This is only as good as the reference matching in your collection. Web of Science references carry a
structured form that matches reliably. Scopus writes free-text citations without a DOI, so grouping
the variants of the same cited work is approximate there, and a `grep()` on a surname may match more
or fewer entries than it should. Check what `cols` actually selected before trusting step 3.

## On the underlying request

What you were really after is a way to go from an edge of a network back to the documents that
produced it, and there is no such path in Biblioshiny today -- which is why you ended up trying to
export the network. That is a reasonable thing to want, and a more useful feature request than this
thread is as a question. If you would like to see it, please open a separate issue describing it that
way: from a selected pair of nodes, list the documents behind the link.
